ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -317,9 +326,48 @@ export const WorkspaceEdit = fp.flow(withCurrentWorkspace(), withCdrVersions(),
       history.back();
     }
 
+    async initialBillingAccountLoad() {
+      // If user hasn't grant GCP billing scope to workbench, we can not fetch billing account from Google
+      // or fetch user's available billing accounts.
+      // When creating/duplicating workspace, show free tier billing account.
+      // When editing existing workspace, show free tier if that is currently being used or 'User Provided Billing Account'
+      // if it is user's billing account.
+      if (serverConfigStore.get().config.enableBillingUpgrade && !hasBillingScope()) {
+        const freeTierBillingAccount: BillingAccount = {
+          name: 'billingAccounts/' + serverConfigStore.get().config.freeTierBillingAccountId,
+          isFreeTier: true,
+          isOpen: true,
+          displayName: 'Use All of Us initial credits',
+        };
+        if (this.isMode(WorkspaceEditMode.Create) || this.isMode(WorkspaceEditMode.Duplicate)) {
+          this.setState(prevState => fp.set(
+              ['workspace', 'billingAccountName'],
+            freeTierBillingAccount.name,
+            prevState));
+          this.setState({'billingAccounts': [freeTierBillingAccount]});
+        } else if (this.isMode(WorkspaceEditMode.Edit)) {
+          // If the user hasn't grant billing scope to workbench yet, keep the server's current value for
+          // billingAccountName and add a shim entry into billingAccounts so the dropdown entry is not empty.
+          //
+          // The server will not perform an updateBillingInfo call if the received billingAccountName
+          // is the same as what is currently stored.
+          if (this.props.workspace.billingAccountName === freeTierBillingAccount.name) {
+            this.setState({'billingAccounts': [freeTierBillingAccount]});
+          } else {
+            this.setState({'billingAccounts': [{
+              name: this.props.workspace.billingAccountName,
+              displayName: 'User Provided Billing Account',
+              isFreeTier: false,
+              isOpen: true}]});
+          }
+        }
+      } else {
+        await this.fetchBillingAccounts();
```

Comment:
so if `enableBillingUpgrade=false`, we will call `fetchBillingAccounts` - is that correct? Does it only return the free tier account if the feature is disabled?